### PR TITLE
Add to README

### PR DESCRIPTION
**Pull request by glacials**
Had no idea what this was when I tripped across it, so wrote it down 😃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,44 @@
 # Narwhal!
+
+[![GoDoc shield](https://raw.githubusercontent.com/golang/gddo/c782c79e0a3c3282dacdaaebeff9e6fd99cb2919/gddo-server/assets/status.svg)][docs]
```
Comment (zombiezen) [inline README.md:3]: pkg.go.dev (the new, not-deprecated site) has badges:

```suggestion
[![Go Reference](https://pkg.go.dev/badge/github.com/yourbase/narwhal?tab=doc)](https://pkg.go.dev/github.com/yourbase/narwhal?tab=doc)
```

File: README.md [hunk 2]
```diff
@@ -1 +1,44 @@
...
+
+Narwhal is a Go API for Docker and Docker containers.
+
+## Examples
```
Comment (zombiezen) [inline README.md:7]: These would be better as [examples](https://pkg.go.dev/testing?tab=doc#hdr-Examples) since they show up in the reference documentation and are tested to ensure they compile. The convention is to use a file `example_test.go` in the same package, but start it with `package narwhal_test` to force the code to import the package.

**Comment by glacials** [comment]
@zombiezen I've moved the examples out of the README and into proper Go examples, but for ease of understanding the repo I do think something is missing from the README again. Personally, I feel that example usage in a GitHub README helps me immediately understand the value of the package, while example usage on pkg.go.dev helps me get my code right during implementation.

If the former is missing I generally won't get much farther than that before looking for alternatives, especially if the repo isn't popular. Open to thoughts.

**Comment by zombiezen** [comment]
I agree about the README. I'm attributing this to a lack of defined scope for this package. Other than few utility functions, it's a pretty thin wrapper over the underlying Docker client. I don't think the abstraction carries its weight.

I do think there are some genuinely useful, common patterns that we could be factoring out from our projects. However, that's far beyond the scope of this PR and needs some dedicated design time that doesn't seem high enough priority right now.

**Comment by beholders-eye** [comment]
IMO, we could start, in the near future, thinking of using Docker's official Go SDK:  https://docs.docker.com/develop/sdk/

go-dockerclient authors recommends it: https://github.com/fsouza/go-dockerclient#difference-between-go-dockerclient-and-the-official-sdk.

**Comment by beholders-eye** [comment]
Or maybe change Narwhal into an abstraction layer on top of Docker's Go SDK and other optional "docker like" (isolation)  implementations out there.

**Comment by glacials** [comment]
¯\\\_(ツ)\_/¯ I'll leave it to y'all what to do in the grand scheme here as my ties to this repo are super light -- all I needed out of this PR was to make it easier for others to figure out what this repo does than it was for me, as I arrived here through a dependency to a big blank page that just said

> # Narwhal!

and had to dig to learn anything more. I'm going to add some really basic usage info on the README and not let this get blocked anymore.

I'll also use this soapbox to bring up my preference for ELI5 project names 😉
